src/ax/endpoint/retry.go

```diff
@@ -1,10 +1,55 @@
 package endpoint
 
-// RetryPolicy returns true if the message should be retried.
-type RetryPolicy func(InboundEnvelope, error) bool
+import (
+	"math"
+	"time"
+)
 
-// DefaultRetryPolicy is a RetryPolicy that rejects a message after it has been
-// attempted three (3) times.
-func DefaultRetryPolicy(env InboundEnvelope, _ error) bool {
-	return env.DeliveryCount < 3
+// RetryPolicy returns is a function responsible for determining whether or not
+// a message should be retried.
+//
+// It returns the delay that occur before retrying, and a bool indicating
```

Comment:
This feels like it needs something, like "*should* occur" or "occur**s**" (with 's') ?
> the delay that *should* occur before retrying